path_secrets.go

```diff
@@ -0,0 +1,94 @@
+package kubesecrets
+
+import (
+	"context"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const secretsPrefix = "secret_name"
+
+func pathSecrets(b *secretsReaderBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: framework.MatchAllRegex(secretsPrefix),
+
+		Fields: map[string]*framework.FieldSchema{
+			"secret_name": {
+				Type:        framework.TypeString,
+				Description: "Specifies the name of the kubernetes secret.",
+				Query:       true,
+				Required:    true,
+			},
+			"namespace": {
+				Type:        framework.TypeString,
+				Description: "Specifies the name of the kubernetes secret namespace.",
+				Query:       true,
+				Required:    true,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: b.handleRead,
+		},
+		HelpDescription: pathInvalidHelp,
+		ExistenceCheck:  b.handleExistenceCheck,
+	}
+}
+
+func (b *secretsReaderBackend) handleRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	secretName := data.Get("secret_name").(string)
+	namespace := data.Get("namespace").(string)
+	b.Logger().Info("In handleRead() secretName: " + secretName + ", namespace: " + namespace)
```

Comment:
I'm lost. Can you explain how this works? (or link to vault docs I can read)
The example you gave is `vault read kubernetes-secrets-reader/my-secret namespace=default`, is the path "kubernetes-secrets-reader/my-secret namespace=default" and "secret_name" key is implicit here?

Reply:
secret_name is defined in Pattern field and thus is part of the url (https://github.com/mesh-for-data/vault-plugin-secrets-kubernetes-reader/blob/1e8251eb63b9e1a03fb0df6115536a25944425da/path_secrets.go#L15)
Vault's code has good explanation: https://github.com/hashicorp/vault/blob/master/sdk/framework/path.go#L49

Reply:
Cool thanks. Why did you use matchall and not genericName?

Reply:
Cool thanks. Why did you use matchall and not genericName?

Reply:
I missed this function... will use it.